src/Microsoft.DotNet.Build.Tasks.Feed/BlobFeedAction.cs

```diff
@@ -89,6 +99,23 @@ public BlobFeedAction(string expectedFeedUrl, string accountKey, MSBuild.TaskLog
                     catch (InvalidOperationException ex) when (ex.Message.Contains("init"))
                     {
                         Log.LogWarning($"Sub-feed {source.FeedSubPath} has not been initialized. Initializing now...");
+                        requiresInit = true;
+                    }
+                    catch (InvalidOperationException ex) when (ex.Message.Contains("Unable to obtain a lock on the feed."))
+                    {
+                        Log.LogWarning($"Sleet was not able to get a lock on the feed. Sleeping {delay} seconds and retrying.");
+
+                        // Pushing packages might take more than just 60 seconds, so on each iteration we multiply the defined delay to a random factor
+                        // Using the defaults this could range from 30 seconds to 12.5 minutes.
```

Comment:
12.5 minutes seems excessive for potentially the first retry. The more I think about this, the more I like the exponential approach intuitively (perhaps with a random sleep time reduction).

Reply:
Sorry, looks like I need to work on my comment, top delay amount on each iteration is 2.5 minutes. If we use the default 5 retried this adds up to 12.5